dango/bank/src/execute.rs

```diff
@@ -206,12 +252,39 @@ pub fn bank_execute(ctx: SudoCtx, msg: BankMsg) -> StdResult<Response> {
     if !recipient_exists {
         ORPHANED_TRANSFERS.may_update(ctx.storage, (msg.from, msg.to), |coins| {
             let mut coins = coins.unwrap_or_default();
-            coins.insert_many(msg.coins)?;
+            coins.insert_many(msg.coins.clone())?;
             Ok::<_, StdError>(coins)
         })?;
     }
 
-    Ok(Response::new())
+    Response::new()
+        .may_add_event(if recipient_exists {
+            Some(Sent {
+                user: msg.from,
+                to: msg.to,
+                coins: msg.coins.clone(),
+            })
+        } else {
+            None
+        })?
+        .may_add_event(if recipient_exists {
+            Some(Received {
+                user: msg.to,
+                from: msg.from,
+                coins: msg.coins.clone(),
+            })
+        } else {
+            None
+        })?
+        .may_add_event(if !recipient_exists {
+            Some(TransferOrphaned {
+                from: msg.from,
+                to: msg.to,
+                coins: msg.coins,
+            })
+        } else {
+            None
+        })
```

Comment:
idk if there's a cleaner syntax for doing this

Reply:
we can have a better syntax if `ContractEvent::new` takes a `&U` instead of `U`. let's fix this in a future PR?